main.go

```diff
@@ -351,40 +352,43 @@ func main() { //nolint:funlen,maintidx
 	}
 }
 
-func createSecretCacheConfig(platform cluster.Platform) map[string]cache.Config {
+func createSecretCacheConfig(ctx context.Context, cli client.Client) map[string]cache.Config {
 	namespaceConfigs := map[string]cache.Config{
 		"istio-system":      {}, // for both knative-serving-cert and default-modelregistry-cert,as an easy workarond, to watch all in this namespace for now
 		"openshift-ingress": {},
 	}
-	switch platform {
-	case cluster.ManagedRhoai:
-		namespaceConfigs["redhat-ods-monitoring"] = cache.Config{}
-		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
-		operatorNs, err := cluster.GetOperatorNamespace()
-		if err != nil {
-			operatorNs = "redhat-ods-operator" // fall back case
-		}
-		namespaceConfigs[operatorNs] = cache.Config{}
-	case cluster.SelfManagedRhoai:
-		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
-	default:
-		namespaceConfigs["opendatahub"] = cache.Config{}
+	// TODO: if we want to not harcoded above two namespace we can add them with label selector
+	// maistra.io/member-of=istio-system
+	// network.openshift.io/policy-group=ingress
+
+	labelSelector := client.MatchingLabels{
+		"opendatahub.io/generated-namespace": "true",
+	}
+	namespaceList := &corev1.NamespaceList{}
+	if err := cli.List(ctx, namespaceList, labelSelector); err != nil {
```

Comment:
on a fresh install, is this going to work ? my understading is that the application namespace is created by the DSCI controller so at the time this `createSecretCacheConfig`  is invoked, then the namespace my not be there yet

Reply:
you are right, this only works for upgrade case, 
clean installation wont get these namespaces